### ISSUE TITLE
[QUESTION] Behavioural Cloning Loss/Entropy Bounds

### ISSUE THREAD

**Issue by cristinagatt** [HumanCompatibleAI/imitation#424]
Hello!

I am using Behavioral Cloning on a sample data set with around 275 samples on a custom Gym Environment. I connected the model to Tensor Board to view the results of the training. Looking at the loss and entropy, they all had values which were bigger than 1 (around 1.12). I trained for quite a big number of epochs so I don't think that more training will decrease the loss. From my experience working with ML, normally the loss is bounded between 0 and 1.

Are the bounds for BC in this library different? Or is the problem I'm having related to the data set or the way I'm training BC?

Thank you in advance.

**Comment by AdamGleave** [comment]
The [loss](https://github.com/HumanCompatibleAI/imitation/blob/master/src/imitation/algorithms/bc.py#L306) used is [cross-entropy](https://en.wikipedia.org/wiki/Cross_entropy) loss (plus entropy bonus and l2 regularization). This doesn't have any upper bound, so 1.12 loss isn't necessarily indicative of a problem, although it does sound fairly high.

It's hard to know what could be going wrong without more context. BC is a fairly inefficient algorithm, so 275 samples may just be too little data for it.

**Comment by AdamGleave** [comment]
Closing due to inactivity, feel free to reopen if you have any other questions.
